### ISSUE TITLE
Always update v1.0 in any new build

### ISSUE THREAD

**Issue by PiPyL** [microsoft/react-native-code-push#2435]
Code push I'm pushing 1.0 and 1.0.1. I built the app with version 1.0.2 and did not push any push code for this 1.0.2 version. But when the app runs, it downloads version 1.0 to update my app version 1.0.2 (both iOS and android). I don't know how to fix it. If anyone knows how to fix it, please guide me. Thank you

**Comment by PiPyL** [comment]
> You can add -t 1.0.2 in command

No. I do not push new version 1.0.2 to code push. I want app version 1.0.2 have not new update version

**Comment by yln05091998** [comment]
i got the same error

**Comment by stefan-schweiger** [comment]
same behavior, released an app with version 1.0.1 and have only codepush updates deployed for 1.0.0 (no additional semver wildcards), but the app always downloads the 1.0.0 update.

When I now release a codepush update for 1.0.1 the app doesn't find those updates. There seems to be some bug with determining the current app version.

**Comment by stefan-schweiger** [comment]
Calling `sync` manually and monitoring network traffic with flipper I see that the correct url is called:

```
GET https://codepush.appcenter.ms/v0.1/public/codepush/update_check?deployment_key=<key>&app_version=1.0.1&client_unique_id=<id>

{
	"update_info": {
		"download_url": "https://codepush.blob.core.windows.net/storagev2/<storageid>",
		"is_available": true,
		"is_disabled": false,
		"target_binary_range": "1.0.1",
		"package_hash": "<hash>.",
		"label": "v11",
		"package_size": 6253135,
		"update_app_version": false,
		"should_run_binary_version": false,
		"is_mandatory": true
	}
}
```

But the update is actually never downloaded and applied.
